modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQConnection.java

```diff
@@ -610,14 +618,53 @@ public boolean attemptReconnection()
         return false;
     }
 
-    public ProtocolVersion makeBrokerConnection(BrokerDetails brokerDetail) throws IOException, AMQException
+    public ProtocolVersion makeBrokerConnection(final BrokerDetails brokerDetail) throws IOException, AMQException
     {
-        return _delegate.makeBrokerConnection(brokerDetail);
+        // Requires permission java.lang.RuntimePermission "modifyThreadGroup"
+        Object returnObject = AccessController.doPrivileged(new PrivilegedAction() {
+            public Object run() {
+                try {
+                    return _delegate.makeBrokerConnection(brokerDetail);
+                } catch (IOException | AMQException e) {
+                    return e;
+                }
+            }
+        });
+
+        if (returnObject instanceof ProtocolVersion) {
+            return (ProtocolVersion) returnObject;
+        } else if (returnObject instanceof IOException) {
```

Comment:
Why do we need to check if returned object is of type "IOException"? Can _delegate.makeBrokerConnection(brokerDetail) return IOException?